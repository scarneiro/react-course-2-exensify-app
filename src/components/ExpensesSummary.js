import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <div className='page-header'>
        <div className='content-container'>
            {
                props.expenseCount === 0 ? (
                        <h1 className='page-header__title'>Viewing no expenses</h1>
                ) : (
                    props.expenseCount === 1 ? (
                        <h1 className='page-header__title'>Viewing <span>{props.expenseCount}</span> expense totalling <span>{numeral(props.expensesTotal / 100).format('$0,0.00')}</span> </h1> 
                    ) : (
                        <h1 className='page-header__title'>Viewing <span>{props.expenseCount}</span> expenses totalling <span>{numeral(props.expensesTotal / 100).format('$0,0.00')}</span> </h1> 
                    )
                )
            }
            <div className='page-header__actions'>
                <Link className="button" to="/create">Add Expense</Link>
            </div>
        </div>
    </div>
);

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expensesTotal: selectExpensesTotal(visibleExpenses),
        expenseCount: visibleExpenses.length
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
