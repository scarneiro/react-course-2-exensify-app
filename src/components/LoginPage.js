import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

export class LoginPage extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = {
            userid: '',
            password: ''
        };
        this.startLogin = props.startLogin
    }
    onSubmit(e) {
        e.preventDefault();
    }
    render() {
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    <div>
                        <label>User</label>
                        <input type="text" placeholder="User Id" />
                    </div>
                    <div>
                        <label>Password</label>
                        <input type="password" placeholder="Password" />
                    </div>
                    <div>
                        <button onClick={this.props.startLogin}>Login</button>
                    </div>
                </form>
            </div>
        )
    };   
};

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);