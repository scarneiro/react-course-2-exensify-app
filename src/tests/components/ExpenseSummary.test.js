import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import expenses from "../fixtures/expenses";
import expensesTotal from '../../selectors/expenses-total';

test('should render component correctly for 1 expense.', () => {
    const total = expensesTotal([ expenses[1] ]);
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[1]]} expensesTotal={total} expenseCount={1} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render component correctly for multiple expenses', () => {
    const total = expensesTotal(expenses);
    const wrapper = shallow(<ExpensesSummary expenses={expenses} expensesTotal={total} expenseCount={expenses.length} />);
    expect(wrapper).toMatchSnapshot();
});