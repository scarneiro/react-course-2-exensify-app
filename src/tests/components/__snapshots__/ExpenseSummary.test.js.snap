// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly for 1 expense. 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       expense totalling 
      <span>
        $1,095.00
      </span>
       
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render component correctly for multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       expenses totalling 
      <span>
        $1,141.95
      </span>
       
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
